### DISCUSSION TITLE
@wagmi/core 1.0.x autoConnect fails to reconnect

### DISCUSSION THREAD

**Discussion by kenkunz** [wevm/wagmi#2439] [Q&A]
I am implementing `@wagmi/core` within a SvelteKit application ([TradingStrategy.ai](https://tradingstrategy.ai/)). I'm currently trying to upgrade from `0.10.10` to `1.0.7`. I have not been able to get `autoConnect` to work with the new version.

The following code works as expected in `0.10.10`:

```javascript
const { chains, provider, webSocketProvider } = configureChains(
	[arbitrum, avalanche, bsc, mainnet, polygon],
	[w3mProvider({ projectId }), publicProvider()]
);

createClient({
	autoConnect: true,
	connectors: w3mConnectors({ projectId, version: 1, chains }),
	provider,
	webSocketProvider
});

console.log(getAccount());
```

After upgrading to `1.0.7`, the code changed minimally (method/prop name changes) to:

```javascript
const { chains, publicClient, webSocketPublicClient } = configureChains(
	[arbitrum, avalanche, bsc, mainnet, polygon],
	[w3mProvider({ projectId }), publicProvider()]
);

createConfig({
	autoConnect: true,
	connectors: w3mConnectors({ projectId, version: 1, chains }),
	publicClient,
	webSocketPublicClient
});

console.log(getAccount());
```

I can see that various `localStorage` key-values are stored. But when I refresh the page, the wallet fails to auto-reconnect.

The following `localStorage` keys are created in `1.0.7`: `wagmi.store`, `wagmi.connected`, `wagmi.wallet`. The same ones are created in `0.10.10`, plus an additional one: `wagmi.injected.shimDisconnect`. I tried clearing all browser application data after switching to `1.0.7` in case this was due to some conflict in format – this didn't help. I've also tried calling `config.autoConnect()` manually – no luck there either. I've tested with multiple injected wallets (MetaMask, Rabby) – same result.

I carefully reviewed the docs it appears that I'm doing everything correctly. Did something about the `autoConnect` API change? Does this seem like a bug? (If yes, I can create a minimal repro and open a bug ticket).

**Comment by skplunkerin** [comment]
I've seen this same issue in a project I'm working on that just updated `wagmi` and made these changes. This seems like a bug to me, but since I can't find many people affected by it or proposes solutions I'm not sure?

**Reply by skplunkerin** [reply]
the best (yet irrelevant) proposed fix I can find is from this question: https://github.com/wagmi-dev/wagmi/discussions/402; but my code already has the proposed solution, yet I still have this problem.

**Reply by kenkunz** [reply]
Thanks for chiming in, @skplunkerin!

I think the issue in #402 is somewhat specific to Next / React, whereas I am only using `@wagmi/core`. In any case, I am also hoisting my `createConfig` call to a context where it is called before any of my components are used.

**Comment by kenkunz** [comment]
**Interesting discovery:** if I begin on my older branch using `0.10.10`, connect my wallet, then switch to my `1.x` branch, autoConnect _works_. If I then clear my application data and connect again (on `1.x` branch), autoConnect _doesn't_ work.

I compared the data stored in `localStorage` between the two versions and found an interesting difference: the older version sets the key `wagmi.injected.shimDisconnect` to `true` (when using `InjectedConnector`); the `1.x` version does not set this key at all.

I tried manually setting this key when connecting with `InjectedConnector` and this seems to fix this issue. I will try creating a minimal repro today to confirm whether this happens outside of my specific SvelteKit use case. If confirmed, I'll open a bug ticket.

CC: @skplunkerin (in case this workaround helps)

**Reply by skplunkerin** [reply]
interesting, thanks for sharing! I'll test this workaround.

**Reply by skplunkerin** [reply]
Dude, this workaround worked for me too, thanks!
🤔 This definitely feels like it's a bug then if we're needing to set this ourselves; either the new version is supposed to set this _(and obviously doesn't)_, or it's supposed to do something else in-place of it and isn't _(while fortunately(?) still able to use this local storage key/value from the old version)_.

**Reply by kenkunz** [reply]
Glad this worked for you as well!

I got sidetracked yesterday and didn't get a chance to try with a minimal repro. Will get to that this morning and open a bug ticket.

**Comment by kenkunz** [comment]
@skplunkerin here's the bug ticket: #2511 – please add any additional context and up-vote.

Thanks!

**Reply by skplunkerin** [reply]
Done, and thank you!
